src/Core/Presentation/Controller/ResetPasswordController.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace App\Core\Presentation\Controller;
+
+use App\Core\Domain\Model\User;
+use App\Core\Presentation\Form\ChangePasswordFormType;
+use App\Core\Presentation\Form\ResetPasswordRequestFormType;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use SymfonyCasts\Bundle\ResetPassword\Controller\ResetPasswordControllerTrait;
+use SymfonyCasts\Bundle\ResetPassword\Exception\ResetPasswordExceptionInterface;
+use SymfonyCasts\Bundle\ResetPassword\ResetPasswordHelperInterface;
+
+#[Route('/reset-password')]
+class ResetPasswordController extends AbstractController
+{
+    use ResetPasswordControllerTrait;
+
+    public function __construct(
+        private ResetPasswordHelperInterface $resetPasswordHelper,
+        private EntityManagerInterface $entityManager
+    ) {}
+
+    /**
+     * Display & process form to request a password reset.
+     */
+    #[Route('', name: 'app_forgot_password_request')]
```

Comment:
Come mai la `route` è vuota? Questo non fa sì che venga intercettata anche se vado su una URL "pulita" del tipo "http://localhost/" ?

Reply:
la rotta eredita da quella impostata nella classe, quindi in questo caso è `/reset-password/`. Nel metodi dopo infatti gli aggiunge altro, tipo in `checkEmail` è `/reset-password/check-email`